models/issue_xref.go

```diff
@@ -273,3 +287,40 @@ func (comment *Comment) RefIssueIdent() string {
 	// FIXME: check this name for cross-repository references (#7901 if it gets merged)
 	return "#" + com.ToStr(comment.RefIssue.Index)
 }
+
+// __________      .__  .__ __________                                     __
+// \______   \__ __|  | |  |\______   \ ____  ________ __   ____   _______/  |_
+//  |     ___/  |  \  | |  | |       _// __ \/ ____/  |  \_/ __ \ /  ___/\   __\
+//  |    |   |  |  /  |_|  |_|    |   \  ___< <_|  |  |  /\  ___/ \___ \  |  |
+//  |____|   |____/|____/____/____|_  /\___  >__   |____/  \___  >____  > |__|
+//                                  \/     \/   |__|           \/     \/
+
+// ResolveCrossReferences will return the list of references to close/reopen by this PR
+func (pr *PullRequest) ResolveCrossReferences() ([]*Comment, error) {
+	unfiltered := make([]*Comment, 0, 5)
+	if err := x.
+		Where("ref_repo_id = ? AND ref_issue_id = ?", pr.Issue.RepoID, pr.Issue.ID).
+		In("ref_action", []references.XRefAction{references.XRefActionCloses, references.XRefActionReopens}).
+		OrderBy("id").
+		Find(&unfiltered); err != nil {
```

Comment:
Can't we do the filter at the SQL level?

Reply:
I guess I could try, but I don't think it's worth the extra effort from xorm/the db. Since duplicates will be the odd cases, this will be much faster for a real-world average of 0~3 results. Unless we're aiming for compact code on our side.

Reply:
@zeripath Having said that, I couldn't find an easy way to do that in xorm. The query in SQL should look like:

```
select * from comment
where id in ( select id from
        ( select issue_id, max(id) as id from comment where ..... group by issue_id )
    )
```

But I don't know how to do it.

Reply:
I guess I was wondering about using group by or distinct but you're right it needs the in.

I think if we're genuinely not expecting lots of duplicates then simply filtering outside of the db is reasonable to avoid an unruly sql monster that would likely take even longer to run.